### PR TITLE
Coverage component and data loading

### PR DESCRIPTION
**Pull request by collijk**

### CODE REVIEW

File: src/vivarium_gates_shigella_vaccine/components/vaccine_coverage.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+    def __init__(self):
+        pass
+
+    @property
+    def name(self):
+        return 'shigella_vaccine_coverage'
+
+    def setup(self, builder):
+        self.coverage = {
```
Comment (KjellSwedin) [inline src/vivarium_gates_shigella_vaccine/components/vaccine_coverage.py:29]: Nit: formatting makes this hard to read

**Review by KjellSwedin — APPROVED** [review]

File: src/vivarium_gates_shigella_vaccine/components/vaccine_coverage.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            dose: builder.lookup.build_table(coverage, key_columns=['sex'], parameter_columns=['age', 'year'])
```
Comment (halfhorst) [inline src/vivarium_gates_shigella_vaccine/components/vaccine_coverage.py]: if coverage doesn't vary by age or sex, why include them in the lookup table?

File: src/vivarium_gates_shigella_vaccine/components/vaccine_coverage.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        }
+
+        columns = [
+            'vaccine_dose',
+            'vaccine_event_time',
+        ]
+        self.population_view = builder.population.get_view(['age', 'alive'] + columns)
+        builder.population.initializes_simulants(self.on_initialize_simulants,
+                                                 creates_columns=columns,
+                                                 requires_streams=['dose_age'])
```
Comment (halfhorst) [inline src/vivarium_gates_shigella_vaccine/components/vaccine_coverage.py]: ~I don't see where this dependency comes in to play~ edit: found it in a later PR

Comment (halfhorst) [reply src/vivarium_gates_shigella_vaccine/components/vaccine_coverage.py]: found it in a later pr

**Review by halfhorst — APPROVED** [review]

**Pull request merged**